### PR TITLE
sx126x fsk

### PR DESCRIPTION
**Pull request by jlpoltrack**
Methods to allow for FSK on SX126x, could enable something like a 50 Hz mode.

### CODE REVIEW

**Comment by olliw42** [comment]
interesting
wouldn't this be a massive drop in range? in my mind it's organized like FSK < FLRC < LoRa. Wouldn't 2.4 GHz when just have all advantages on its side?
Am I assuming correctly that you have some bench test?

**Comment by jlpoltrack** [comment]
Per the datasheet, even at 250 kbps (which I think would be enough for the 111 Hz mode), sensitivity is spec'd at -104 dBm (same as the FLRC mode used) + there would be another ~8 dB by using 868/915 vs 2.4:

image

No bench tests yet.

Edit - Just for reference, the calculator has a bug when calculating Time on Air - someone forgot to adjust for ms.  To calculate you can use Bits / Real Bitrate, example 835 / 250000 * 1000 = 3.34 ms.

image

File: src/sx126x.h [hunk 1]
```diff
@@ -504,6 +510,77 @@ typedef enum {
     SX126X_OCP_CONFIGURATION_140_MA       = 0x38,
 } SX126X_OCP_CONFIG_ENUM;
 
+//-------------------------------------------------------
+// Enum Definitions FSK
```
Comment (olliw42) [inline src/sx126x.h]: can you pl change to GFSK

**Comment by olliw42** [comment]
I'm can's see if it's using tabs or 4 spaces. If tabs can you pl ensure it's all 4 spaces (tabs can be replaced with spaces e.g. with Notepoad++)

File: src/sx126x.h (hunk 1 above)
Comment (jlpoltrack) [reply src/sx126x.h]: Updated

**Comment by jlpoltrack** [comment]
> I'm can's see if it's using tabs or 4 spaces. If tabs can you pl ensure it's all 4 spaces (tabs can be replaced with spaces e.g. with Notepoad++)

Just checked and only spaces are used.  VSCode automatically replaces a tab with 4 spaces.

**Comment by olliw42** [comment]
MANY THX!
let's see where this is leadijg too :D:D
mergining and closing

**Pull request merged**